README.md

```diff
@@ -0,0 +1,225 @@
+# ObserveFS - DuckDB Filesystem Observability Extension
+
+
+## What is ObserveFS?
+
+ObserveFS is a powerful DuckDB extension that provides comprehensive **filesystem observability** for your data operations. It transparently wraps httpfs (HTTP, S3, Hugging Face) with monitoring capabilities, giving you detailed insights into I/O performance, latency patterns, and usage metrics.
+
+Whether you're optimizing data pipelines, debugging performance issues, or understanding access patterns, ObserveFS gives you the visibility you need.
+
+## Usage
+
+```sql
+-- Install and load the ObserveFS extension
+FORCE INSTALL observefs;
+LOAD observefs;
+
+-- Query remote data (automatically monitored)
+SELECT count(*) FROM 'https://huggingface.co/datasets/open-r1/OpenR1-Math-220k/resolve/main/data/train-00003-of-00010.parquet';
+
+-- View detailed performance metrics
+SELECT observefs_get_profile();
+
+-- Clear metrics for fresh analysis
+SELECT observefs_clear();
+```
+
+The output includes comprehensive metrics:
+- Operation-specific latency histograms (READ, OPEN, LIST)
+- Quantile analysis (P50, P75, P90, P95, P99)
+- Per-bucket performance breakdown
+- Min/Max/Mean latency statistics
+
+## Getting started
+Clone this repository using:
+```sh
+git clone --recurse-submodules https://github.com/dentiny/duckdb-filesystem-observability.git
+```
+Note that `--recurse-submodules` will ensure DuckDB is pulled which is required to build the extension.
+
+## Building
```

Comment:
Development guide should be placed in another doc, or at the very bottom of the readme

Reply:
Duckdb extensions are very user centric, not developer oriented